### ISSUE TITLE
Simplify Bitbucket Server plugin interaction with Sourcegraph

### ISSUE THREAD

**Issue by kzh** [sourcegraph/sourcegraph#7824]
There are some improvements that can be made to the existing plugin interaction between Bitbucket Server and Sourcegraph.

**Webhooks**
We can remove the unnecessary REST API to CRUD webhooks. There only needs to be a single webhook (that exist by default) to dispatch event payloads that are helpful to automation. This does not have to be configurable to the extent of changing event scope, name, etc.

The webhook can be toggled (on/off + secret) on the Bitbucket Server Sourcegraph admin page. The secret here will be used to HMAC sign payloads. The secret configured on Sourcegraph will have to match this one.

**Configuration**
Clean up the BBS configuration on Sourcegraph. It is currently fragmented between the experimental site administration config (for faster permissions) and BBS external service (webhook secret). Ideally, everything will be moved to the BBS extsvc on a per instance basis.
```
"plugin": {
  "webhook": {
     "secret": "secret"
  },
  "acl_endpoints": {
    "enabled": "true"
  }
} 
```

**Version** 
The BBS plugin will expose a HTTP endpoint (`/version`) that the Sourcegraph instance can ping to discover which features are enabled/available.

This endpoint would return something like:
```
{
   "version": "1.2",
   "features: ["acl", "automation-webhook"],
}
```

**Comment by tsenart** [comment]
I don't believe this could be only 1d of work. Let's be more conservative.

**Comment by uwedeportivo** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.13 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by tsenart** [comment]
@kzh: Wasn't there a PR open for this? What made you change the milesone?

**Comment by kzh** [comment]
The PR didn't make the 3.13 branch cut so I changed this tracking issue to 3.14 to indicate that it'll be released in 3.14. Also, I'm not sure if the other parts of this issue are still under consideration (the version negotiation).
